File: app/src/main/java/com/sudoajay/duplication_data/Receive_Boot_Completed/MyRebootReceiver.java
package com.sudoajay.duplication_data.Receive_Boot_Completed;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.sudoajay.duplication_data.sharedPreferences.TraceBackgroundService;

public class MyRebootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {

        TraceBackgroundService traceBackgroundService = new TraceBackgroundService(context);
        if (!traceBackgroundService.isBackgroundServiceWorking()) {
            Intent serviceIntent = new Intent(context, ForegroundServiceBoot.class);
            serviceIntent.setAction("RebootReceiver");
            context.startService(serviceIntent);
        }
    }
}

File: app/src/main/java/com/sudoajay/duplication_data/receiveBootCompleted/MyRebootReceiver.kt
package com.sudoajay.duplication_data.receiveBootCompleted

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.sudoajay.duplication_data.sharedPreferences.TraceBackgroundService

class MyRebootReceiver : BroadcastReceiver() {
    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    override fun onReceive(context: Context, intent: Intent) {
        val traceBackgroundService = TraceBackgroundService(context)
        if (!traceBackgroundService.isBackgroundServiceWorking) {
            val serviceIntent = Intent(context, ForegroundServiceBoot::class.java)
            serviceIntent.action = "RebootReceiver"
            context.startService(serviceIntent)
        }
    }
}